packages/mui-lab/src/TreeItem/TreeItem.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { TreeItem as XTreeItem, TreeItemProps } from '@mui/x-tree-view/TreeItem';
+
+let warnedOnce = false;
+
+const warn = () => {
+  if (!warnedOnce) {
+    console.warn(
+      [
+        'MUI: The TreeItem component was moved from `@mui/lab` to `@mui/x-tree-view`.',
+        'The component will no longer be exported from `@mui/lab` in the first release of October 2023.',
+        '',
+        "You should use `import { TreeItem } from '@mui/x-tree-view'`",
+        "or `import { TreeItem } from '@mui/x-tree-view/TreeItem'`",
+        '',
+        'More information about this migration on our blog: INSERT BLOG POST.',
```

Comment:
I will not merge this PR before adding the blog post link here (and on the `TreeView` component)